compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -319,11 +322,8 @@
     private static final ParserRuleContext[] PARAM_END =
             { ParserRuleContext.COMMA, ParserRuleContext.CLOSE_PARENTHESIS };
 
-    private static final ParserRuleContext[] STMT_START_WITH_EXPR_RHS = { ParserRuleContext.ASSIGN_OP,
-            ParserRuleContext.RIGHT_ARROW, ParserRuleContext.COMPOUND_BINARY_OPERATOR, ParserRuleContext.SEMICOLON };
-
-    private static final ParserRuleContext[] EXPR_STMT_RHS = { ParserRuleContext.SEMICOLON, ParserRuleContext.ASSIGN_OP,
-            ParserRuleContext.RIGHT_ARROW, ParserRuleContext.COMPOUND_BINARY_OPERATOR };
+    private static final ParserRuleContext[] STMT_START_WITH_EXPR_RHS = { ParserRuleContext.ASSIGNMENT_STMT_RHS,
+            ParserRuleContext.RIGHT_ARROW, ParserRuleContext.SEMICOLON };
```

Comment:
Shall we keep both `STMT_START_WITH_EXPR_RHS` and `EXPR_STMT_RHS`. Apparently, the original test case added to test this functionality is now outdated.

We can modify [call_stmt_source_07.bal](https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-parser/src/test/resources/statements/call-stmt/call_stmt_source_07.bal) to,
```bal
class Person {

    function foo() {
        bar() // semicolon should recover here
        someString.indexOf() // semicolon should recover here
    }
}

function bar() {
    someString.indexOf() // semicolon should recover here
    condition = condition - 1;
}
```

Relevant logic in the parser:
https://github.com/ballerina-platform/ballerina-lang/blob/f4b37dc14c266caeaf671b2354e71601e4a23580/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java#L8347-L8352